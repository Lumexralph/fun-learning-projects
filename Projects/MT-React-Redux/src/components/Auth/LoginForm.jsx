import React from 'react';
import PropTypes from 'prop-types';

const LoginForm = ({
  onUsernameChange,
  onPasswordInput,
  onSubmit,
}) => (
  <div>
    <form
      onSubmit={event => onSubmit(event)}
    >
      <div className="container">
        <div className="popup">
          <span className="popuptext" id="myPopup">Ensure the information are correct..</span>
        </div>
        <label htmlFor="uname">
          <input
            onChange={event => onUsernameChange(event)}
            type="text"
            placeholder="enter your username"
            name="uname"
            required
          />
        </label>

        <label htmlFor="psw">
          <input
            onChange={event => onPasswordInput(event)}
            type="password"
            placeholder="enter your password"
            name="psw"
            required
          />
        </label>
        <p>Welcome back, ready to serve you when you&apos;re in</p>

        <button id="loginButton" className="btn signup-btn background-tertiary" type="submit">Login to FixZit</button>

      </div>

    </form>
  </div>
);

LoginForm.propTypes = {
  onUsernameChange: PropTypes.func.isRequired,
  onPasswordInput: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};


export default LoginForm;
