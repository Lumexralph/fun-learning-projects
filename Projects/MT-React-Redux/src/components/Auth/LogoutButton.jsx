import clearStorageData from '../../helpers/clearStorageData';

const logOut = (history) => {
  clearStorageData();

  history.push('/');

  return null;
};

export default logOut;
