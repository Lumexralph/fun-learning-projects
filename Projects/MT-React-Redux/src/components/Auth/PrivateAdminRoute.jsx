import React from 'react';
import { Redirect, Route } from 'react-router-dom';

import getDataFromStorage from '../../helpers/getDataFromStorage';

const PrivateAdminRoute = ({ component: Component, ...rest }) => {
  const admin = getDataFromStorage('admin');

  return (
    <Route
      {...rest}
      render={(props) => {
        if (admin === 'true') {
          return (<Component {...props} />);
        }
        return (
          <Redirect
            to={{
              pathname: '/login',
              state: { from: props.location },
            }}
          />
        );
      }
      }
    />);
};

export default PrivateAdminRoute;
