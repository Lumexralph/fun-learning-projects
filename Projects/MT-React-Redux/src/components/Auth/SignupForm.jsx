import React from 'react';
import PropTypes from 'prop-types';

const SignupForm = ({
  onSubmit,
  onUsernameChange,
  onEmailChange,
  onPasswordChange,
}) => (
  <div className="right-part">
    <form onSubmit={event => onSubmit(event)}>
      <div className="container">
        <div className="popup">
          <span className="popuptext" id="myPopup" />
        </div>
        <label htmlFor="username">
          <input
            onChange={event => onUsernameChange(event)}
            type="text"
            placeholder="pick a username"
            name="username"
            pattern="[a-zA-Z]+"
            title="Please, provide username with only letters."
            required
          />
        </label>


        <label htmlFor="email">
          <input
            onChange={event => onEmailChange(event)}
            type="email"
            placeholder="your email address"
            name="email"
            required
          />
        </label>


        <label htmlFor="psw1">
          <input
            onChange={event => onPasswordChange(event)}
            type="password"
            placeholder="create a password"
            name="psw1"
            minLength="8"
            pattern="^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?!.*\s).*$"
            title="Please include at least 1 uppercase character, 1 lowercase character, and 1 number."
            required
          />
        </label>


        <label htmlFor="psw2">
          <input
            onChange={event => onPasswordChange(event)}
            type="password"
            placeholder="confirm password"
            name="psw2"
            minLength="8"
            pattern="^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?!.*\s).*$"
            title="Please include at least 1 uppercase character, 1 lowercase character, and 1 number."
            required
          />
        </label>

        <p>Use at least 1 uppercase, 1 lowercase character, and 1 number</p>

        <button
          className="btn signup-btn background-tertiary"
          type="submit"
        >Sign up for FixZit
        </button>
      </div>
    </form>
  </div>
);

SignupForm.propTypes = {
  onUsernameChange: PropTypes.func.isRequired,
  onEmailChange: PropTypes.func.isRequired,
  onPasswordChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};


export default SignupForm;
