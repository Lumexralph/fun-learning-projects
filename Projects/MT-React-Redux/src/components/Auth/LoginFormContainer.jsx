import React from 'react';
import PropTypes from 'prop-types';

import LoginForm from './LoginForm';

const LoginFormContainer = ({
  onUsernameChange,
  onPasswordInput,
  onSubmit,
}) => (
  <main className="signup-container-img">
    <section className="signup-container login">
      <LoginForm
        onUsernameChange={onUsernameChange}
        onPasswordInput={onPasswordInput}
        onSubmit={onSubmit}
      />
    </section>
  </main>
);

LoginFormContainer.propTypes = {
  onUsernameChange: PropTypes.func.isRequired,
  onPasswordInput: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default LoginFormContainer;
