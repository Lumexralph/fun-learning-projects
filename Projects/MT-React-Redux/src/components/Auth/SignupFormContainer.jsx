import React from 'react';
import PropTypes from 'prop-types';

import SignupForm from './SignupForm';

const SignupFormContainer = ({
  onSubmit,
  onUsernameChange,
  onEmailChange,
  onPasswordChange,
}) => (
  <main className="signup-container-img">
    <section className="signup-container">
      <SignupForm
        onSubmit={onSubmit}
        onUsernameChange={onUsernameChange}
        onEmailChange={onEmailChange}
        onPasswordChange={onPasswordChange}
      />
    </section>
  </main>
);

SignupFormContainer.propTypes = {
  onUsernameChange: PropTypes.func.isRequired,
  onEmailChange: PropTypes.func.isRequired,
  onPasswordChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default SignupFormContainer;
