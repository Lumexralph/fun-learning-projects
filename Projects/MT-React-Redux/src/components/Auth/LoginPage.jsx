import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import HeaderContainer from '../home/HeaderContainer';
import FooterContainer from '../home/FooterContainer';
import LoginFormContainer from './LoginFormContainer';
import loginUser from '../../actions/loginUser';
import currentUser from '../../actions/currentUser';
import ShowLoadingStatus from '../LoadingStatus';
import updateStorageData from '../../helpers/updateStorageData';

export class LoginPage extends Component {
  state = {
    navText: ['Home', 'Signup'],
    username: '',
    password: '',
    isLoading: false,
    message: null,
  }

  handleUsernameChange = (event) => {
    const { target } = event;
    this.setState({ username: target.value });
  }

  handlePasswordInput = (event) => {
    const { target } = event;
    this.setState({ password: target.value });
  }

  handleSubmit = (event) => {
    event.preventDefault();
    const { login, setCurrentUser, history } = this.props;
    const { username, password } = this.state;

    this.setState({ isLoading: true });
    login({ username, password })
      .then((response) => {
        setCurrentUser(response.data);
        updateStorageData('token', response.data.token);
        updateStorageData('admin', response.data.adminRole);
        this.setState({
          isLoading: false,
        });
        if (response.data.adminRole) {
          history.push('/admin');
        } else {
          history.push('/profile');
        }
      })
      .catch((error) => {
        const errorMessage = (!error.response) ? 'Please ensure you have internet connection' : error.response.data.message;

        this.setState({
          isLoading: false,
          message: errorMessage,
        });

        return errorMessage;
      });
  }

  render() {
    const { navText, message, isLoading } = this.state;
    return (
      <div className="container">
        <HeaderContainer navText={navText} />
        <ShowLoadingStatus
          status={isLoading}
          text={message}
        />
        <LoginFormContainer
          onUsernameChange={this.handleUsernameChange}
          onPasswordInput={this.handlePasswordInput}
          onSubmit={this.handleSubmit}
        />
        <FooterContainer />
      </div>
    );
  }
}

LoginPage.propTypes = {
  login: PropTypes.func.isRequired,
  setCurrentUser: PropTypes.func.isRequired,
  history: PropTypes.object.isRequired,
};


export const mapDispatchToProps = dispatch => ({
  login: user => dispatch(loginUser(user)),
  setCurrentUser: user => dispatch(currentUser(user)),
});


export default connect(null, mapDispatchToProps)(LoginPage);
