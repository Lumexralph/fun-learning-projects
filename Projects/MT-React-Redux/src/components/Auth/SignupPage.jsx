import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';

import HeaderContainer from '../home/HeaderContainer';
import SignupFormContainer from './SignupFormContainer';
import FooterContainer from '../home/FooterContainer';
import registerUser from '../../actions/registerUser';
import currentUser from '../../actions/currentUser';
import ShowLoadingStatus from '../LoadingStatus';
import updateStorageData from '../../helpers/updateStorageData';


export class SignupPage extends Component {
  state = {
    navText: ['Home', 'Login'],
    username: '',
    email: '',
    password1: '',
    password2: '',
    message: null,
    isLoading: false,
  }

  handleUsernameChange = (event) => {
    const { target } = event;
    this.setState({ username: target.value });
  }

  handleEmailChange = (event) => {
    const { target } = event;
    this.setState({ email: target.value });
  }

  handlePasswordChange = (event) => {
    const { target } = event;
    const { name, value } = target;

    this.setState((previousState) => {
      let { password1, password2 } = previousState;

      if (name === 'psw1') {
        password1 = value;
        return { password1 };
      }

      password2 = value;
      return { password2 };
    });
  }

  handleSubmit = (event) => {
    const {
      password1, password2, username, email,
    } = this.state;

    this.setState({ isLoading: true });

    const { signupNewUser, setCurrentUser } = this.props;
    event.preventDefault();
    const result = (password1 === password2);

    if (result) {
      const user = {
        password1, password2, username, email,
      };

      signupNewUser(user)
        .then((response) => {
          setCurrentUser(response.data);
          updateStorageData('token', response.data.token);
          updateStorageData('admin', response.data.adminRole);
          this.setState({
            message: 'Signup successful',
            isLoading: false,
          });
        })
        .catch((error) => {
          const errorMessage = (!error.response) ? 'Please ensure you have internet connection' : error.response.data.message;

          this.setState({
            isLoading: false,
            message: errorMessage,
          });

          return errorMessage;
        });
    } else {
      this.setState({
        isLoading: false,
        message: 'Password do not match',
      });
    }
  }

  render() {
    const { navText, isLoading, message } = this.state;

    return (
      <div className="container">
        {message === 'Signup successful' ? <Redirect to="/profile" /> : null}
        <HeaderContainer navText={navText} />
        <ShowLoadingStatus
          status={isLoading}
          text={message}
        />
        <SignupFormContainer
          onSubmit={this.handleSubmit}
          onUsernameChange={this.handleUsernameChange}
          onEmailChange={this.handleEmailChange}
          onPasswordChange={this.handlePasswordChange}
          buttonStatus={isLoading}
        />
        <FooterContainer />
      </div>
    );
  }
}


SignupPage.propTypes = {
  signupNewUser: PropTypes.func.isRequired,
  setCurrentUser: PropTypes.func.isRequired,
};

export const mapDispatchToProps = dispatch => ({
  signupNewUser: user => dispatch(registerUser(user)),
  setCurrentUser: user => dispatch(currentUser(user)),
});

export default connect(null,
  mapDispatchToProps)(SignupPage);
