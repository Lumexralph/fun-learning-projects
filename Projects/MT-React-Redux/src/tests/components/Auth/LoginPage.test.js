import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import { LoginPage, mapDispatchToProps } from '../../../components/Auth/LoginPage';

describe('Test for LoginPage with admin', () => {
  const response = {
    data: {
      adminRole: true,
    },
  };

  const login = Promise.resolve(response);

  const component = renderer.create(
    <Router>
      <LoginPage
        login={() => login}
        setCurrentUser={e => e}
        history={{}}
      />
    </Router>
    ,
  );
  let tree = component.toJSON();
  const componentInstance = component.root;

  it('Should render component', () => {
    expect(tree).toMatchSnapshot();
  });

  it('Should call submit event on form', (done) => {
    componentInstance.findByType('form').props.onSubmit({
      preventDefault: () => true,
    });

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    done();
  });

  it('Should have the onChange event on input field', (done) => {
    componentInstance.findByProps({ name: 'uname' }).props.onChange({ target: { value: '' } });
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    done();
  });

  it('Should have the onChange event on input field', (done) => {
    componentInstance.findByProps({ name: 'psw' }).props.onChange({ target: { value: '' } });
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    done();
  });

  it('Should dispatch login action', (done) => {
    const dispatch = jest.fn();

    expect(mapDispatchToProps(dispatch).login({ name: 'test' })).toBeUndefined();
    done();
  });

  it('Should dispatch setCurrentUser action', (done) => {
    const dispatch = jest.fn();

    expect(mapDispatchToProps(dispatch).setCurrentUser({ name: 'test' })).toBeUndefined();
    done();
  });
});
