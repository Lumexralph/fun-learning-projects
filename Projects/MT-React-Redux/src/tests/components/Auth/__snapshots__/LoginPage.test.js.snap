// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for LoginPage with admin Should call submit event on form 1`] = `
<div
  className="container"
>
  <header>
    <img
      alt="header logo"
      className="header-logo"
      src="test-file-stub"
    />
    <nav>
      <ul
        className="nav"
      >
        <li>
          <a
            className="btn nav-btn nav-link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            className="btn nav-btn nav-link"
            href="/signup"
            onClick={[Function]}
          >
            Signup
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <div
    className="loader"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 32 32\\">
  <circle transform=\\"translate(8 0)\\" cx=\\"0\\" cy=\\"16\\" r=\\"0\\"> 
    <animate attributeName=\\"r\\" values=\\"0; 4; 0; 0\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0\\"
      keytimes=\\"0;0.2;0.7;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.6 0.4 0.8\\" calcMode=\\"spline\\" />
  </circle>
  <circle transform=\\"translate(16 0)\\" cx=\\"0\\" cy=\\"16\\" r=\\"0\\"> 
    <animate attributeName=\\"r\\" values=\\"0; 4; 0; 0\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.3\\"
      keytimes=\\"0;0.2;0.7;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.6 0.4 0.8\\" calcMode=\\"spline\\" />
  </circle>
  <circle transform=\\"translate(24 0)\\" cx=\\"0\\" cy=\\"16\\" r=\\"0\\"> 
    <animate attributeName=\\"r\\" values=\\"0; 4; 0; 0\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.6\\"
      keytimes=\\"0;0.2;0.7;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.6 0.4 0.8\\" calcMode=\\"spline\\" />
  </circle>
</svg>
",
        }
      }
      style={
        Object {
          "fill": "lightblue",
          "height": "5%",
          "width": "20%",
        }
      }
    />
  </div>
  <main
    className="signup-container-img"
  >
    <section
      className="signup-container login"
    >
      <div>
        <form
          onSubmit={[Function]}
        >
          <div
            className="container"
          >
            <div
              className="popup"
            >
              <span
                className="popuptext"
                id="myPopup"
              >
                Ensure the information are correct..
              </span>
            </div>
            <label
              htmlFor="uname"
            >
              <input
                name="uname"
                onChange={[Function]}
                placeholder="enter your username"
                required={true}
                type="text"
              />
            </label>
            <label
              htmlFor="psw"
            >
              <input
                name="psw"
                onChange={[Function]}
                placeholder="enter your password"
                required={true}
                type="password"
              />
            </label>
            <p>
              Welcome back, ready to serve you when you're in
            </p>
            <button
              className="btn signup-btn background-tertiary"
              id="loginButton"
              type="submit"
            >
              Login to FixZit
            </button>
          </div>
        </form>
      </div>
    </section>
  </main>
  <footer
    className="footer background-quarternary"
  >
    <div>
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="app logo"
          className="footer-logo"
          src="test-file-stub"
        />
      </a>
    </div>
    <div>
      <div
        className="footer-details text-secondary"
      >
        <div
          className="footer-info"
        >
          <div>
            FixZit 5555 Address St. #555 City, State 55555
          </div>
          <div>
            01-2245-789
          </div>
          <div
            className="footer-icon"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              <i
                className="fa fa-facebook-f text-secondary"
              />
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              <i
                className="fa fa-linkedin text-secondary"
              />
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              <i
                className="fa fa-twitter text-secondary"
              />
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              <i
                className="fa fa-instagram text-secondary"
              />
            </a>
          </div>
        </div>
        <div
          className="footer-links"
        >
          <div
            className="footer-links-copyright"
          >
            Copyright © 2018 ALL RIGHTS RESERVED
          </div>
          <div
            className="footer-nav-links-container"
          >
            <ul
              className="footer-nav-links"
            >
              <li>
                <a
                  href="/"
                  onClick={[Function]}
                >
                  ABOUT US
                </a>
              </li>
              <li>
                <a
                  href="/"
                  onClick={[Function]}
                >
                  PRIVACY
                </a>
              </li>
              <li>
                <a
                  href="/"
                  onClick={[Function]}
                >
                  TERMS OF USE
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </footer>
</div>
`;

exports[`Test for LoginPage with admin Should have the onChange event on input field 1`] = `
<div
  className="container"
>
  <header>
    <img
      alt="header logo"
      className="header-logo"
      src="test-file-stub"
    />
    <nav>
      <ul
        className="nav"
      >
        <li>
          <a
            className="btn nav-btn nav-link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            className="btn nav-btn nav-link"
            href="/signup"
            onClick={[Function]}
          >
            Signup
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <div
    className="loader-text"
  >
    <p>
      Please ensure you have internet connection
    </p>
  </div>
  <main
    className="signup-container-img"
  >
    <section
      className="signup-container login"
    >
      <div>
        <form
          onSubmit={[Function]}
        >
          <div
            className="container"
          >
            <div
              className="popup"
            >
              <span
                className="popuptext"
                id="myPopup"
              >
                Ensure the information are correct..
              </span>
            </div>
            <label
              htmlFor="uname"
            >
              <input
                name="uname"
                onChange={[Function]}
                placeholder="enter your username"
                required={true}
                type="text"
              />
            </label>
            <label
              htmlFor="psw"
            >
              <input
                name="psw"
                onChange={[Function]}
                placeholder="enter your password"
                required={true}
                type="password"
              />
            </label>
            <p>
              Welcome back, ready to serve you when you're in
            </p>
            <button
              className="btn signup-btn background-tertiary"
              id="loginButton"
              type="submit"
            >
              Login to FixZit
            </button>
          </div>
        </form>
      </div>
    </section>
  </main>
  <footer
    className="footer background-quarternary"
  >
    <div>
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="app logo"
          className="footer-logo"
          src="test-file-stub"
        />
      </a>
    </div>
    <div>
      <div
        className="footer-details text-secondary"
      >
        <div
          className="footer-info"
        >
          <div>
            FixZit 5555 Address St. #555 City, State 55555
          </div>
          <div>
            01-2245-789
          </div>
          <div
            className="footer-icon"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              <i
                className="fa fa-facebook-f text-secondary"
              />
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              <i
                className="fa fa-linkedin text-secondary"
              />
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              <i
                className="fa fa-twitter text-secondary"
              />
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              <i
                className="fa fa-instagram text-secondary"
              />
            </a>
          </div>
        </div>
        <div
          className="footer-links"
        >
          <div
            className="footer-links-copyright"
          >
            Copyright © 2018 ALL RIGHTS RESERVED
          </div>
          <div
            className="footer-nav-links-container"
          >
            <ul
              className="footer-nav-links"
            >
              <li>
                <a
                  href="/"
                  onClick={[Function]}
                >
                  ABOUT US
                </a>
              </li>
              <li>
                <a
                  href="/"
                  onClick={[Function]}
                >
                  PRIVACY
                </a>
              </li>
              <li>
                <a
                  href="/"
                  onClick={[Function]}
                >
                  TERMS OF USE
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </footer>
</div>
`;

exports[`Test for LoginPage with admin Should have the onChange event on input field 2`] = `
<div
  className="container"
>
  <header>
    <img
      alt="header logo"
      className="header-logo"
      src="test-file-stub"
    />
    <nav>
      <ul
        className="nav"
      >
        <li>
          <a
            className="btn nav-btn nav-link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            className="btn nav-btn nav-link"
            href="/signup"
            onClick={[Function]}
          >
            Signup
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <div
    className="loader-text"
  >
    <p>
      Please ensure you have internet connection
    </p>
  </div>
  <main
    className="signup-container-img"
  >
    <section
      className="signup-container login"
    >
      <div>
        <form
          onSubmit={[Function]}
        >
          <div
            className="container"
          >
            <div
              className="popup"
            >
              <span
                className="popuptext"
                id="myPopup"
              >
                Ensure the information are correct..
              </span>
            </div>
            <label
              htmlFor="uname"
            >
              <input
                name="uname"
                onChange={[Function]}
                placeholder="enter your username"
                required={true}
                type="text"
              />
            </label>
            <label
              htmlFor="psw"
            >
              <input
                name="psw"
                onChange={[Function]}
                placeholder="enter your password"
                required={true}
                type="password"
              />
            </label>
            <p>
              Welcome back, ready to serve you when you're in
            </p>
            <button
              className="btn signup-btn background-tertiary"
              id="loginButton"
              type="submit"
            >
              Login to FixZit
            </button>
          </div>
        </form>
      </div>
    </section>
  </main>
  <footer
    className="footer background-quarternary"
  >
    <div>
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="app logo"
          className="footer-logo"
          src="test-file-stub"
        />
      </a>
    </div>
    <div>
      <div
        className="footer-details text-secondary"
      >
        <div
          className="footer-info"
        >
          <div>
            FixZit 5555 Address St. #555 City, State 55555
          </div>
          <div>
            01-2245-789
          </div>
          <div
            className="footer-icon"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              <i
                className="fa fa-facebook-f text-secondary"
              />
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              <i
                className="fa fa-linkedin text-secondary"
              />
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              <i
                className="fa fa-twitter text-secondary"
              />
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              <i
                className="fa fa-instagram text-secondary"
              />
            </a>
          </div>
        </div>
        <div
          className="footer-links"
        >
          <div
            className="footer-links-copyright"
          >
            Copyright © 2018 ALL RIGHTS RESERVED
          </div>
          <div
            className="footer-nav-links-container"
          >
            <ul
              className="footer-nav-links"
            >
              <li>
                <a
                  href="/"
                  onClick={[Function]}
                >
                  ABOUT US
                </a>
              </li>
              <li>
                <a
                  href="/"
                  onClick={[Function]}
                >
                  PRIVACY
                </a>
              </li>
              <li>
                <a
                  href="/"
                  onClick={[Function]}
                >
                  TERMS OF USE
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </footer>
</div>
`;

exports[`Test for LoginPage with admin Should render component 1`] = `
<div
  className="container"
>
  <header>
    <img
      alt="header logo"
      className="header-logo"
      src="test-file-stub"
    />
    <nav>
      <ul
        className="nav"
      >
        <li>
          <a
            className="btn nav-btn nav-link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            className="btn nav-btn nav-link"
            href="/signup"
            onClick={[Function]}
          >
            Signup
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    className="signup-container-img"
  >
    <section
      className="signup-container login"
    >
      <div>
        <form
          onSubmit={[Function]}
        >
          <div
            className="container"
          >
            <div
              className="popup"
            >
              <span
                className="popuptext"
                id="myPopup"
              >
                Ensure the information are correct..
              </span>
            </div>
            <label
              htmlFor="uname"
            >
              <input
                name="uname"
                onChange={[Function]}
                placeholder="enter your username"
                required={true}
                type="text"
              />
            </label>
            <label
              htmlFor="psw"
            >
              <input
                name="psw"
                onChange={[Function]}
                placeholder="enter your password"
                required={true}
                type="password"
              />
            </label>
            <p>
              Welcome back, ready to serve you when you're in
            </p>
            <button
              className="btn signup-btn background-tertiary"
              id="loginButton"
              type="submit"
            >
              Login to FixZit
            </button>
          </div>
        </form>
      </div>
    </section>
  </main>
  <footer
    className="footer background-quarternary"
  >
    <div>
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="app logo"
          className="footer-logo"
          src="test-file-stub"
        />
      </a>
    </div>
    <div>
      <div
        className="footer-details text-secondary"
      >
        <div
          className="footer-info"
        >
          <div>
            FixZit 5555 Address St. #555 City, State 55555
          </div>
          <div>
            01-2245-789
          </div>
          <div
            className="footer-icon"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              <i
                className="fa fa-facebook-f text-secondary"
              />
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              <i
                className="fa fa-linkedin text-secondary"
              />
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              <i
                className="fa fa-twitter text-secondary"
              />
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              <i
                className="fa fa-instagram text-secondary"
              />
            </a>
          </div>
        </div>
        <div
          className="footer-links"
        >
          <div
            className="footer-links-copyright"
          >
            Copyright © 2018 ALL RIGHTS RESERVED
          </div>
          <div
            className="footer-nav-links-container"
          >
            <ul
              className="footer-nav-links"
            >
              <li>
                <a
                  href="/"
                  onClick={[Function]}
                >
                  ABOUT US
                </a>
              </li>
              <li>
                <a
                  href="/"
                  onClick={[Function]}
                >
                  PRIVACY
                </a>
              </li>
              <li>
                <a
                  href="/"
                  onClick={[Function]}
                >
                  TERMS OF USE
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
